src/ui/components/filters/facetscomponent.js

```diff
@@ -240,25 +240,72 @@ export default class FacetsComponent extends Component {
   }
 
   setState (data) {
-    const previousFacets = data['filters'] || [];
-    let updatedFacets = [];
+    const facets = data['filters'] || [];
+    let processedFacets = this._processFacets(facets);
 
     if (this._transformFacets) {
-      const facetsCopy = cloneDeep(previousFacets);
-      const transformedFacets = this._transformFacets(facetsCopy, this.config);
-
-      updatedFacets = transformedFacets;
-    } else {
-      updatedFacets = previousFacets;
+      const facetsCopy = cloneDeep(facets);
+      processedFacets = this._transformFacets(facetsCopy, this.config);
     }
 
     return super.setState({
       ...data,
-      filters: Facet.fromCore(updatedFacets),
+      filters: Facet.fromCore(processedFacets),
       isNoResults: data.resultsContext === ResultsContext.NO_RESULTS
     });
   }
 
+  /**
+   * Applies default formatting to different types of facets
+   *
+   * @param {DisplayableFacet[]} facets from answers-core
+   * @returns {DisplayableFacet[]} from answers-core
+   */
+  _processFacets (facets) {
+    const processedFacets = [];
+    const booleanFacets = [];
+
+    const isBooleanFacet = facet => {
+      const firstOption = (facet.options && facet.options[1]) || {};
+      return firstOption['value'] === true || firstOption['value'] === false;
+    };
+
+    facets.forEach(facet => {
+      if (isBooleanFacet(facet)) {
+        booleanFacets.push(facet);
+      } else {
+        processedFacets.push(facet);
+      }
```

Comment:
is it possible to avoid these pushes by changing transformBooleanFacets() to take in a single facet instead of an array, and using a map instead of forEach on line 273?

Reply:
sure, I like that better